Source/Charts/Renderers/BarChartRenderer.swift

```diff
@@ -399,9 +396,104 @@ open class BarChartRenderer: BarLineScatterCandleBubbleRenderer
+    {
+
+        guard let gradientPositions = dataSet.gradientPositions else
+        {
+            assertionFailure("Must set `gradientPositions if `dataSet.drawBarGradientEnabled` is true")
+            return
+        }
+
+        guard let boundingBox = buffer.rects.union() else { return }
+        guard !boundingBox.isNull, !boundingBox.isInfinite, !boundingBox.isEmpty else { return }
+
+        let drawBorder = dataSet.barBorderWidth > 0
+
+        let gradientStart = dataSet.gradientEnd.isInfinite ?
+            CGPoint(x: boundingBox.minX, y: boundingBox.minY) :
+            dataSet.gradientEnd.applying(matrix)
+
+        let gradientEnd = dataSet.gradientStart.isInfinite ?
+            CGPoint(x: boundingBox.minX, y: boundingBox.maxY) :
+            dataSet.gradientStart.applying(matrix)
+
+        var gradientColorComponents: [CGFloat] = []
+        var gradientLocations: [CGFloat] = []
+
+        for position in gradientPositions.reversed()
+        {
+            let location = CGPoint(x: boundingBox.minX, y: position)
+                .applying(matrix)
+            let normalizedLocation =
+                (location.y - boundingBox.minY) / (boundingBox.maxY - boundingBox.minY)
+            switch normalizedLocation {
+            case ..<0:
+                gradientLocations.append(0)
+            case 0..<1:
+                gradientLocations.append(normalizedLocation)
+            case 1...:
+                gradientLocations.append(1)
+            default:
+                assertionFailure()
+            }
+        }
+
+        for color in dataSet.colors.reversed()
+        {
+            guard let (r, g, b, a) = color.nsuirgba else {
+                continue
+            }
+            gradientColorComponents += [r, g, b, a]
+        }
+
+        let baseColorSpace = CGColorSpaceCreateDeviceRGB()
+        guard let gradient = CGGradient(
+            colorSpace: baseColorSpace,
+            colorComponents: &gradientColorComponents,
+            locations: &gradientLocations,
+            count: gradientLocations.count) else {
+                return
+        }
+
+        for barRect in buffer.rects
+        {
+            context.saveGState()
+            defer { context.restoreGState() }
+
+            guard viewPortHandler.isInBoundsLeft(barRect.maxX) else { continue }
+            guard viewPortHandler.isInBoundsRight(barRect.minX) else { break }
+
+            context.beginPath()
+            context.addRect(barRect)
+            context.clip()
+            context.drawLinearGradient(gradient, start: gradientStart, end: gradientEnd, options: [])
+
+            if drawBorder
+            {
+                context.setStrokeColor(dataSet.barBorderColor.cgColor)
+                context.setLineWidth(dataSet.barBorderWidth)
+                context.stroke(barRect)
+            }
+        }
```

Comment:
in gradient bars, `accessibilityOrderedElements` is missing?

Reply:
@liuxuan30 pushed commit. Hope that I did it correct.